bundles/CoreBundle/Request/ParamConverter/DataObjectParamConverter.php

```diff
@@ -55,7 +56,9 @@ public function apply(Request $request, ParamConverter $configuration)
         if (!$object) {
             throw new NotFoundHttpException(sprintf('Invalid data object ID given for parameter "%s".', $param));
         } elseif (!$object->isPublished()) {
-            throw new NotFoundHttpException(sprintf('Data object for parameter "%s" is not published.', $param));
+            if (!(Tool::isFrontendRequestByAdmin() && Tool\Authentication::authenticateSession($request))) {
```

Comment:
@dvesh3 Please also consider the workspace permissions for the logged in user (`$user->isAllowed('view')`).